### PR TITLE
[NEW] Add Option to view user profile avatar in fullscreen

### LINKED ISSUE #2042: [NEW] Option to view user profile avatar in fullscreen.
## Description

There should be an option to view user profile avatar in full screen when you click it in profileFragment and userDetailFragment.

### PR DESCRIPTION
**Pull request by GOVINDDIXIT**
@RocketChat/android

Closes #2042 

#### Changes: 
User can view his profile picture on full screen
User can view profile pictures of other users on full screen

#### Screenshots or GIF for the change:
GIF-190501_014007

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt [hunk 1]
```diff
@@ -226,6 +226,16 @@ class ProfileFragment : Fragment(), ProfileView, ActionMode.Callback {
             hideUpdateAvatarOptions()
             presenter.resetAvatar()
         }
+
+        button_view_profile_image.setOnClickListener {
+            hideUpdateAvatarOptions()
+            val avatarUrl = presenter.getImageUri()
+            chat.rocket.android.profile.ui.newInstance(avatarUrl)
+            fragmentManager?.beginTransaction()
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt]: You should move this code for our navigator class

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by GOVINDDIXIT** [comment]
@filipedelimabrito I have made the requested changes

**Comment by GOVINDDIXIT** [comment]
@filipedelimabrito Pls review.  I have also updated the GIF

File: app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomNavigator.kt [hunk 2]
```diff
@@ -156,6 +157,11 @@ class ChatRoomNavigator(internal val activity: ChatRoomActivity) {
         activity.overridePendingTransition(R.anim.open_enter, R.anim.open_exit)
     }
 
+    fun toProfileImageDialog(avatarUrl: String) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomNavigator.kt]: `toProfileImage` is enough - the "same" way we have declaring other functions here.

File: app/src/main/java/chat/rocket/android/main/presentation/MainNavigator.kt [hunk 3]
```diff
@@ -48,6 +49,12 @@ class MainNavigator(internal val activity: MainActivity) {
         }
     }
 
+    fun toProfileImageDialog(avatarUrl: String) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/main/presentation/MainNavigator.kt]: `toProfileImage` is enough - the "same" way we have declaring other functions here.

File: app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt [hunk 4]
```diff
@@ -246,6 +246,12 @@ class ProfileFragment : Fragment(), ProfileView, ActionMode.Callback {
             hideUpdateAvatarOptions()
             presenter.resetAvatar()
         }
+
+        button_view_profile_image.setOnClickListener {
+            hideUpdateAvatarOptions()
+            val avatarUrl = presenter.getImageUri()
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt]:
```
with(presenter) {
  toProfileImage(getImageUri())
}
```

File: app/src/main/res/layout/image_dialog_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:background="@color/color_black">
+
+    <com.facebook.drawee.view.SimpleDraweeView
```
Comment (philipbrito) [inline app/src/main/res/layout/image_dialog_fragment.xml]: Can we make use of the old and good `imageView` here? Then use it with `glide`

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by GOVINDDIXIT** [comment]
Hi @filipedelimabrito I have updated the PR with the requested changes, I have also added caching support and placeholder to the image. Pls review

File: app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt [hunk 6]
```diff
@@ -213,6 +213,10 @@ class ProfilePresenter @Inject constructor(
         }
     }
 
+    fun getImageUri(): String {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt]: Actually this function is useless.

File: app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import com.bumptech.glide.Glide
+import com.bumptech.glide.Priority
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.image_dialog_fragment.*
+
+private lateinit var avatarUrl: String
+internal const val TAG_IMAGE_DIALOG_FRAGMENT = "ImageDialogFragment"
+
+fun newInstance(ImageUri: String): Fragment {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt]: It's not an URI. It's an URL!

File: app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    return ImageDialogFragment().apply {
+        avatarUrl = ImageUri
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt]: It also answer my review above.

File: app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt (hunk 7 above)
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt]: Please do not capitalize variables!

Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/profile/ui/ImageDialogFragment.kt]: Could you let us know why you instantiating it this way?

File: app/src/main/res/layout/fragment_user_details.xml [hunk 9]
```diff
@@ -30,7 +30,7 @@
             app:layout_constraintTop_toTopOf="@+id/image_blur" />
 
         <ImageView
-            android:id="@+id/image_avatar"
+            android:id="@+id/user_image_avatar"
```
Comment (philipbrito) [inline app/src/main/res/layout/fragment_user_details.xml]: Please make sure to follow our coding style before changing the things! Wha's the reason to change the id of the imageView here?

File: app/src/main/java/chat/rocket/android/userdetails/ui/UserDetailsFragment.kt [hunk 10]
```diff
@@ -96,7 +103,7 @@ class UserDetailsFragment : Fragment(), UserDetailsView {
         ).into(image_blur)
 
         requestBuilder.apply(RequestOptions.bitmapTransform(RoundedCorners(14)))
-            .into(image_avatar)
+            .into(user_image_avatar)
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/userdetails/ui/UserDetailsFragment.kt]: Please make sure to follow our coding style before changing the things! Wha's the reason to change the id of the imageView here?

File: app/src/main/res/layout/image_dialog_fragment.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:background="@color/color_black">
+
+
+    <ImageView
+        android:id="@+id/detailImage"
```
Comment (philipbrito) [inline app/src/main/res/layout/image_dialog_fragment.xml]: Please make sure to follow our coding style when declaring the widget ids!

File: app/src/main/res/layout/image_dialog_fragment.xml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:background="@color/color_black">
+
```
Comment (philipbrito) [inline app/src/main/res/layout/image_dialog_fragment.xml]: Unnecessary blank line.

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -62,6 +62,7 @@ https://github.com/RocketChat/java-code-styles/blob/master/CODING_STYLE.md#strin
     <string name="action_drawing">Drawing</string>
     <string name="action_save_to_gallery">Save to gallery</string>
     <string name="action_select_photo_from_gallery">Select photo from gallery</string>
+    <string name="action_view_profile_photo">View Profile photo</string>
```
Comment (philipbrito) [inline app/src/main/res/values/strings.xml]: View profile photo instead.

File: app/src/main/res/values-zh-rCN/strings.xml [hunk 14]
```diff
@@ -50,6 +50,7 @@
     <string name="action_drawing">绘画</string>
     <string name="action_save_to_gallery">保存到图库</string>
     <string name="action_select_photo_from_gallery">从图库选照片</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-zh-rCN/strings.xml]: View profile photo instead.

File: app/src/main/res/values-uk/strings.xml [hunk 15]
```diff
@@ -20,6 +20,7 @@
     <string name="title_password">Змінити пароль</string>
     <string name="title_update_profile">Оновити профіль</string>
     <string name="title_create_channel">Створити новий канал</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-uk/strings.xml]: View profile photo instead.

File: app/src/main/res/values-tr/strings.xml [hunk 16]
```diff
@@ -52,6 +52,7 @@
     <string name="action_select_photo_from_gallery">Galeriden resim seç</string>
     <string name="action_take_a_photo">Fotoğraf çek</string>
     <string name="action_reset_avatar">Avatarı kaldır</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-tr/strings.xml]: View profile photo instead.

File: app/src/main/res/values-ru-rRU/strings.xml [hunk 17]
```diff
@@ -54,6 +54,7 @@
     <string name="action_reset_avatar">Восстановить аватар</string>
     <string name="action_connect_server">Соединиться с сервером</string>
     <string name="action_join_community">Сообщество</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-ru-rRU/strings.xml]: View profile photo instead.

File: app/src/main/res/values-pt-rBR/strings.xml [hunk 18]
```diff
@@ -52,6 +52,7 @@
     <string name="action_select_photo_from_gallery">Escolher foto da galeria</string>
     <string name="action_take_a_photo">Tirar uma foto</string>
     <string name="action_reset_avatar">Resetar avatar</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-pt-rBR/strings.xml]: View profile photo instead.

File: app/src/main/res/values-ja/strings.xml [hunk 19]
```diff
@@ -19,6 +19,7 @@
     <string name="title_admin_panel">管理パネル</string>
     <string name="title_password">パスワードの変更</string>
     <string name="title_update_profile">プロフィールの更新</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-ja/strings.xml]: View profile photo instead.

File: app/src/main/res/values-it/strings.xml [hunk 20]
```diff
@@ -48,6 +48,7 @@
     <string name="msg_busy">Occupato</string>
     <string name="msg_invisible">Invisibile</string>
     <string name="action_drawing">Disegno</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-it/strings.xml]: View profile photo instead.

File: app/src/main/res/values-hi-rIN/strings.xml [hunk 21]
```diff
@@ -43,6 +43,7 @@
     <string name="action_attach_a_files">एक फ़ाइल जोडो</string>
     <string name="action_confirm_password">पासवर्ड परिवर्तन की पुष्टि करें</string>
     <string name="action_join_chat">चैट में शामिल हों</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-hi-rIN/strings.xml]: View profile photo instead.

File: app/src/main/res/values-fr/strings.xml [hunk 22]
```diff
@@ -48,6 +48,7 @@
     <string name="msg_busy">Occupé</string>
     <string name="msg_invisible">Invisible</string>
     <string name="action_drawing">Dessin</string>
+    <string name="action_view_profile_photo">View Profile photo</string>
```
Comment (philipbrito) [inline app/src/main/res/values-fr/strings.xml]: View profile photo instead.

File: app/src/main/res/values-fa/strings.xml [hunk 23]
```diff
@@ -48,6 +48,7 @@
     <string name="msg_busy">مشغول</string>
     <string name="msg_invisible">نامرئی</string>
     <string name="action_drawing">کشیدن</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-fa/strings.xml]: View profile photo instead.

File: app/src/main/res/values-es/strings.xml [hunk 24]
```diff
@@ -48,6 +48,7 @@
     <string name="msg_busy">Ocupado</string>
     <string name="msg_invisible">Invisible</string>
     <string name="action_drawing">Dibujo</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-es/strings.xml]: View profile photo instead.

File: app/src/main/res/values-de/strings.xml [hunk 25]
```diff
@@ -48,6 +48,7 @@
     <string name="msg_busy">Beschäftigt</string>
     <string name="msg_invisible">Unsichtbar</string>
     <string name="action_drawing">Zeichnung</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Add translation -->
```
Comment (philipbrito) [inline app/src/main/res/values-de/strings.xml]: View profile photo instead.

File: app/src/main/res/values-ar/strings.xml [hunk 26]
```diff
@@ -57,6 +57,7 @@
     <string name="action_create_server">إنشاء خادم جديد</string>
     <string name="action_register">تسجيل</string>
     <string name="action_confirm">تأكيد</string>
+    <string name="action_view_profile_photo">View Profile photo</string><!-- TODO Translate -->
```
Comment (philipbrito) [inline app/src/main/res/values-ar/strings.xml]: View profile photo instead.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Review by philipbrito — APPROVED** [review]
The requested changes were fixed by me here. Thank you for sending this PR!

**Pull request merged**